### PR TITLE
[ALLUXIO-2119] Centralize version checking test and avoid whitebox

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2119

The version checking functionality is in `AbstractClient`, so it makes sense to test it in `AbstactClientTest`

### CODE REVIEW

**Comment by apc999** [comment]
LGTM

File: core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    public void checkVersion(AlluxioService.Client thriftClient) throws IOException {
+      checkVersion(thriftClient, Constants.BLOCK_MASTER_CLIENT_SERVICE_VERSION);
+    }
+  }
+
+  /**
+   * Tests for an unsupported version.
+   */
+  @Test
+  public void unsupportedVersionTest() throws Exception {
```
Comment (apc999) [inline core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java]: `unsupportedVersion`?

File: core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    protected String getServiceName() {
+      return SERVICE_NAME;
+    }
+
+    @Override
+    protected long getServiceVersion() {
+      return CLIENT_SERVICE_VERSION;
+    }
+
+    public void checkVersion(AlluxioService.Client thriftClient) throws IOException {
```
Comment (apc999) [inline core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java]: it is a bit strange here to hard-code `Constants.BLOCK_MASTER_CLIENT_SERVICE_VERSION` for tests.
How about

```java
public void checkVersionForTest(AlluxioService.Client thriftClient, long version) throws IOException {
    checkVersion(thriftClient, version);
}
```

and then you can have two test cases to test unsupported version with `version=Constants.BLOCK_MASTER_CLIENT_SERVICE_VERSION` and supported version with
`version= CLIENT_SERVICE_VERSION`

File: core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java (hunk 1 above)
Comment (aaudiber) [inline core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java]: Updated, thanks

File: core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java (hunk 2 above)
Comment (aaudiber) [inline core/server/src/test/java/alluxio/worker/block/AbstractClientTest.java]: Good catch, I initially copied from the BlockMasterClientTest and forgot to replace everything specific to BlockMaster. It should all be updated now (I followed your suggestion)

**Comment by aaudiber** [comment]
@apc999 Comments addressed, PTAL

**Pull request merged**